File: programmers/level2/타겟_넘버.java
public class 타겟_넘버 {
	static
	class Solution {
		static int answer = 0;
		static int targetG;
	    public int solution(int[] numbers, int target) {
	    	targetG = target;
	    	dfs(numbers,0,0);
	    	return answer;
	    }
	    void dfs(int[] numbers, int sum, int idx) {
	    	if(idx == numbers.length) {
	    		if(sum == targetG) {
	    			answer++;
	    		}
	    		return;
	    	}
	    	dfs(numbers,sum+numbers[idx], idx+1);
	    	dfs(numbers,sum-numbers[idx], idx+1);
	    }
	}
	public static void main(String[] args) {
		int[] numbers = {1,1,1,1,1};
		int target = 3;
		Solution sol = new Solution();
		System.out.println(sol.solution(numbers, target));
	}
}

File: programmers/python/level2/타겟_넘버.py
from itertools import combinations
def solution(numbers, target):
    answer = 0
    for i in range(1, len(numbers)+1):
        for combi in combinations(range(len(numbers)), i):
            sum = 0
            for j in range(len(numbers)):
                if j in combi:
                    sum += numbers[j]
                else:
                    sum -= numbers[j]
            if sum == target:
                answer += 1
    return answer

print(solution([1,1,1,1,1],3))